pandas/io/sas/sas7bdat.py

```diff
@@ -265,9 +267,11 @@ def _get_properties(self) -> None:
         # Get encoding information
         buf = self._read_bytes(const.encoding_offset, const.encoding_length)[0]
         if buf in const.encoding_names:
-            self.file_encoding = const.encoding_names[buf]
+            self.inferred_encoding = const.encoding_names[buf]
+            if self.encoding == "infer":
+                self.encoding = self.inferred_encoding
         else:
-            self.file_encoding = f"unknown (code={buf})"
+            self.inferred_encoding = f"unknown (code={buf})"
```

Comment:
Aside, it doesn't look like this attribute is used anywhere else?

Reply:
Yes, found that weird as well.
I also chose to rename it because (1) it has no implications to other code (2) reduce confusion, since there is `file_encoding`, `encoding` and `default_encoding`. The name `inferred_encoding` includes the source of this variables content.

Reply:
Okay makes sense. Would take a followup PR to remove the statefullness attributes.